stmtcacher.go

```diff
@@ -18,6 +24,7 @@ type DBProxy interface {
 	Queryer
 	QueryRower
 	Preparer
+	io.Closer
```

Comment:
It isn't generally backward compatible to extend an exported interface like this. Things that previously implemented DBProxy would suddenly not.

Reply:
Very good point. How might we achieve exposing this functionality, if not baked into the interface?

Coupled with you other suggestion, would you suggest something like:
```go
type Clearer interface {
	Clear() error
}

type DBProxy interface {
	Execer
	Queryer
	QueryRower
	Preparer
}

type DBProxyClearer interface {
	DBProxy
	Clearer
)
```

and then the user could do an interface upgrade to clear the statement cache?

Reply:
I just went ahead and implemented my suggestion to aid in discussion.

Reply:
Yeah. Its messy but I think its the only backward compatible solution here.

Reply:
I took a closer look at this. I think we should change `stmtCacher` to `StmtCacher`, have `NewStmtCacher` return `*StmtCacher`, and not create any new interfaces. I wrote this a long time ago and didn't understand Go as well - stmtCacher should have never been unexported.

Reply:
This will make it easier to use - as-is you have to type-assert the return from `NewStmtCacher` to a `Clearer` or `DBProxyClearer` in order to use `Clear`.

Reply:
That would certainly make this cleaner. I would be happy to take a stab at that.

Reply:
@lann do you feel the same way about `stmtCacheProxy` ?

Reply:
One other thought. `StmtCacher` or `StmtCache` I know some are fond of adjective form for interfaces. Does `StmtCacher` sound too much like an interface?

Reply:
I can't remember what its for...

This is one of those cases where you look at code and ask "what idiot wrote this?" and it was you.

Reply:
`StmtCache` sounds good.

Reply:
I experience this frequently 😂

Reply:
Actually, given that `NewStmtCacher` is exported, probably good for that to return `StmtCacher`.

Reply:
Totally 👍

Reply:
Went ahead and export stmtCacher to StmtCacher